In [13]:
import pandas as pd
import matplotlib.pyplot as plt

# Load the dataset
df = pd. read_csv('insurance_claims_cleaned_fixed.csv')

In [14]:
#lets take a look at the incident_type column

df['incident_type'].value_counts()

incident_type
Multi-vehicle Collision     419
Single Vehicle Collision    403
Vehicle Theft                94
Parked Car                   84
Name: count, dtype: int64

In [ ]:
#lets see how incident type affects fraud reported
df.groupby('incident_type')['fraud_reported'].value_counts(normalize = True)



incident_type             fraud_reported
Multi-vehicle Collision   No                0.727924
                          Yes               0.272076
Parked Car                No                0.904762
                          Yes               0.095238
Single Vehicle Collision  No                0.709677
                          Yes               0.290323
Vehicle Theft             No                0.914894
                          Yes               0.085106
Name: proportion, dtype: float64

Multi-vehicle Collision: 27.2% fraud rate

Single Vehicle Collision: 29.0% fraud rate


Key insights:

Collisions (especially single vehicle) have much higher fraud rates.
Parked car and vehicle theft claims are much less likely to be fraudulent.

Explore incident_severity and collision_type for similar patterns.
Visualize these rates (bar plot) for clearer comparison.